src/js/component/app.vue

```diff
@@ -1,12 +1,36 @@
 <template>
     <section id="app" class="app">
-        <top-nav></top-nav>
+        <TopNav/>
         <div class="app__content">
-            <splash></splash>
-            <search></search>
-            <movies-list></movies-list>
+            <Splash v-if="recommendations.type === 'initial'" :is-supported="recorder"/>
+            <Search v-else-if="query" :query="query"/>
+
+            <div v-if="recommendations.type === 'loading'" class="movies-list__loader">
+                <p>Loading movies list...</p>
+            </div>
+            <div v-else-if="recommendation.type === 'failure'">
+                <h1>Oh no :(</h1>
+                <p>{{ recommendation.type.data }}</p>
+            </div>
+
+            <template v-if="recommendations.type === 'success'">
+              <div v-if="recommendations.data.movies.length > 0">
+                <MovieItem
+                  v-for="movie in movies"
+                  :movie="movie"
+                  :key="movie.name"/>
+              </div>
+              <div v-else class="movies-list__empty content">
+                No movies found.
+              </div>
+            </template>
         </div>
-        <recorder></recorder>
+
+        <Recorder
+          v-if="recorder"
+          :is-recording="recommendations.type === 'recording'"
+          @start-recording="startRecording"
+          />
```

Comment:
I like this component to describe the logic and user flow of the application.
The children are good to display data.